sdks/java/io/xml/src/main/java/org/apache/beam/sdk/io/xml/XmlIO.java

```diff
@@ -325,6 +355,13 @@ public boolean matches(String filename) {
       return toBuilder().setCompressionType(compressionType).build();
     }
 
+    /**
+     * Sets the XML file charset.
+     */
+    public Read<T> withCharset(String charset) {
```

Comment:
I'm updating to have the user facing API uses `withCharset(Charset)` and internally use the `String` representation of the `Charset` (for the serialization).